lib/Net/Stripe.pm

```diff
@@ -147,6 +147,16 @@ if your account is using Stripe API version 2015-02-18 or later, you should
 update any code that relies on strict checking of the return value of
 Charge->status.
 
+=item deprecate direct handling of PANs
```

Comment:
If you do decide to go ahead with this, I suggest putting this title in CAPITALS! :-)

Reply:
will do. but i am hoping, fingers crossed, that we don't have any users who are making use of these endpoints. or that if they are, then they are highly in-tune to these kinds of changes and their impact.

Reply:
how about the following format, which will (hopefully) highlight these sections in the table of contents on the CPAN page

```
RELEASE NOTES
  Version 0.40
   BREAKING CHANGES
    deprecate direct handling of PANs
        Stripe strongly discourages direct handling of PANs (primary account
        numbers), even in test mode, and returns invalid_request_error when
        passing PANs to the API from accounts that were created after October
        2017. In live mode, all tokenization should be performed via
        client-side libraries because direct handling of PANs violates PCI
        compliance. So we have removed the methods and parameter constraints
        that allow direct handling of PANs and updated the unit tests
        appropriately.

   BUG FIXES
    fix post_charge() arguments
        Some argument types for `customer` and `card` were non-functional in
        the current code and have been removed from the Kavorka method
        signature. We removed `Net::Stripe::Customer` and `HashRef` for
        `customer` and we removed `Net::Stripe::Card` and `Net::Stripe::Token`
        for `card`, as none of these forms were being serialized correctly for
        passing to the API call. We must retain `Net::Stripe::Card` for the
        `card` attribute in `Net::Stripe::Charge` because the `card` value
        returned from the API call is objectified into a card object. We have
        also added TypeConstraints for the string arguments passed and added
        in-method validation to ensure that the passed argument values make
        sense in the context of one another.

    cleanup post_card():
        Some argument types for `card` are not legitimate, or are being
        deprecated and have been removed from the Kavorka method signature. We
        removed `Net::Stripe::Card` and updated the string validation to
        disallow card id for `card`, as neither of these are legitimate when
        creating or updating a card
        <https://github.com/lukec/stripe-perl/issues/138>, and the code path
        that appeared to be handling `Net::Stripe::Card` was actually
        unreachable <https://github.com/lukec/stripe-perl/issues/100>. We
        removed the dead code paths and made the conditional structure more
        explicit, per discussion in
        <https://github.com/lukec/stripe-perl/pull/133>. We also added unit
        tests for all calling forms, per
        <https://github.com/lukec/stripe-perl/issues/139>.

    fix post_customer() arguments
        Some argument types for `card` are not legitimate and have been
        removed from the Kavorka method signature. We removed
        `Net::Stripe::Card` and updated the string validation to disallow card
        id for `card`, as neither of these are legitimate when creating or
        updating a customer <https://github.com/lukec/stripe-perl/issues/138>.
        We have also updated the structure of the method so that we always
        create a Net::Stripe::Customer object before posting

        <https://github.com/lukec/stripe-perl/issues/148> and cleaned up and
        centralized Net::Stripe:Token coercion code.

   ENHANCEMENTS
    coerce old lists
        In older Stripe API versions, some list-type data structures were
        returned as arrayrefs. We now coerce those old-style lists and
        collections into the hashref format that newer versions of the API
        return, with metadata stored top-level keys and the list elements in
        an arrayref with the key 'data', which is the format that
        "Net::Stripe::List" expects. This makes the SDK compatible with the
        Stripe API back to the earliest documented API version
        <https://stripe.com/docs/upgrades#2011-06-21>.

    add update_card()
        This method allows updates to card address, expiration, metadata, etc
        for existing customer cards.

    encode card metdata in convert_to_form_fields()
        When passing a hashref with a nested metadata hashref to _post(), that
        metadata must be encoded properly before being passed to the Stripe
        API. There is now a dedicated block in convert_to_form_fields for this
        operation. This update was necessary because of the addition of
        update_card(), which accepts a card hashref, which may include
        metadata.

   UPDATES
    update statement_description to statement_descriptor
        The statement_description attribute is now statement_descriptor for
        Net::Stripe::Charge and Net::Stripe::Plan. The API docs
        <https://stripe.com/docs/upgrades#2014-12-17> indicate that this
        change is backwards-compatible. You must update your code to reflect
        this change for parameters passed to these objects and methods called
        on these objects.

    update unit tests for Charge->status
        For Stripe API versions after 2015-02-18
        <https://stripe.com/docs/upgrades#2015-02-18>, the status property on
        the Charge object has a value of 'succeeded' for successful charges.
        Previously, the status property would be 'paid' for successful
        charges. This change does not affect the API calls themselves, but if
        your account is using Stripe API version 2015-02-18 or later, you
        should update any code that relies on strict checking of the return
        value of Charge->status.

SEE ALSO
    <https://stripe.com>, <https://stripe.com/docs/api>
...
```

Reply:
Looks good!